### PR TITLE
[SHELL32] Split dialogs/folder_options.cpp

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

The dialogs/folder_options.cpp source file in shell32 is too big to manage. We will split it to smaller files.
JIRA issue: [CORE-12906](https://jira.reactos.org/browse/CORE-12906)

### CODE REVIEW

File: dll/win32/shell32/dialogs/filetypes.cpp [hunk 1]
```diff
@@ -165,7 +163,8 @@ GetFileTypeIconsByKey(HKEY hKey, PFILE_TYPE_ENTRY Entry)
         return FALSE;
 
     // Get the icon location
-    WCHAR szLocation[MAX_PATH + 32] = { 0 };
+    WCHAR szLocation[MAX_PATH + 32];
+    szLocation[0] = 0;
```
Comment (learn-more) [inline dll/win32/shell32/dialogs/filetypes.cpp]: Why did you change this?

File: dll/win32/shell32/dialogs/filetypes.cpp [hunk 2]
```diff
@@ -180,7 +179,7 @@ GetFileTypeIconsByKey(HKEY hKey, PFILE_TYPE_ENTRY Entry)
 static BOOL
 QueryFileDescription(LPCWSTR ProgramPath, LPWSTR pszName, INT cchName)
 {
-    SHFILEINFOW FileInfo = { 0 };
+    SHFILEINFOW FileInfo;
```
Comment (learn-more) [inline dll/win32/shell32/dialogs/filetypes.cpp]: Why did you remove the initialization?

Comment (katahiromz) [reply dll/win32/shell32/dialogs/filetypes.cpp]: It doesn't use the icon handle in here. And the return value of SHGetFileInfo will be checked. So SHFILEINFO doesn't require initialization.

File: dll/win32/shell32/dialogs/filetypes.cpp (hunk 1 above)
Comment (HBelusca) [reply dll/win32/shell32/dialogs/filetypes.cpp]: I'm not sure it will really make a difference.

Comment (katahiromz) [reply dll/win32/shell32/dialogs/filetypes.cpp]: Revert or not [R/n]?

File: dll/win32/shell32/dialogs/filetypes.cpp (hunk 2 above)
Comment (katahiromz) [reply dll/win32/shell32/dialogs/filetypes.cpp]: Revert or not [R/n]?

File: dll/win32/shell32/dialogs/filetypes.cpp (hunk 1 above)
Comment (HBelusca) [reply dll/win32/shell32/dialogs/filetypes.cpp]: Please revert.

Comment (katahiromz) [reply dll/win32/shell32/dialogs/filetypes.cpp]: OK.

File: dll/win32/shell32/dialogs/filetypes.cpp (hunk 2 above)
Comment (katahiromz) [reply dll/win32/shell32/dialogs/filetypes.cpp]: OK.

**Comment by katahiromz** [comment]
Done.

File: dll/win32/shell32/dialogs/view.cpp [hunk 3]
```diff
@@ -0,0 +1,996 @@
...
+        Item.mask = TVIF_HANDLE | TVIF_IMAGE | TVIF_SELECTEDIMAGE;
+        Item.hItem = pEntry->hItem;
+        Item.iImage = Item.iSelectedImage = ViewTree_GetImage(pEntry);
+        TreeView_SetItem(hwndTreeView, &Item);
+    }
+
+    PropSheet_Changed(GetParent(hwndDlg), hwndDlg);
+}
+
+// FIXME: These macros should not be defined here
```
[the remark this answers is not in the document]
Comment (katahiromz) [reply dll/win32/shell32/dialogs/view.cpp]: OK.

File: dll/win32/shell32/dialogs/view.cpp [hunk 4]
```diff
@@ -0,0 +1,996 @@
...
+        if (lstrcmpiW(pEntry->szKeyName, L"ShowInfoTip") == 0)
+        {
+            pSS->fShowInfoTip = bChecked ? 1 : 0;
+            *pdwMask |= SSF_SHOWINFOTIP;
+            continue;
+        }
+    }
+}
+
+extern "C"
```
[the remark this answers is not in the document]
Comment (katahiromz) [reply dll/win32/shell32/dialogs/view.cpp]: OK.

File: dll/win32/shell32/dialogs/view.cpp [hunk 5]
```diff
@@ -0,0 +1,996 @@
...
+VOID WINAPI SHGetSetSettings(LPSHELLSTATE lpss, DWORD dwMask, BOOL bSet);
+
+static BOOL CALLBACK
+RefreshBrowsersCallback(HWND hWnd, LPARAM msg)
+{
+    WCHAR ClassName[100];
+    if (GetClassName(hWnd, ClassName, 100))
```
[the remark this answers is not in the document]
Comment (katahiromz) [reply dll/win32/shell32/dialogs/view.cpp]: OK.

File: dll/win32/shell32/dialogs/view.cpp [hunk 6]
```diff
@@ -0,0 +1,996 @@
...
+
+    switch (uMsg)
+    {
+        case WM_INITDIALOG:
+            return ViewDlg_OnInitDialog(hwndDlg);
+
+        case WM_COMMAND:
+            switch (LOWORD(wParam))
+            {
+                case 14004: // Restore Defaults
```
[the remark this answers is not in the document]
Comment (katahiromz) [reply dll/win32/shell32/dialogs/view.cpp]: OK.

File: dll/win32/shell32/dialogs/filetypes.cpp [hunk 7]
```diff
@@ -0,0 +1,1780 @@
...
+    if (*pszExt != L'.')
+        return FALSE;
+
+    // open ".ext" key
+    HKEY hKey;
+    if (RegOpenKeyExW(HKEY_CLASSES_ROOT, pszExt, 0, KEY_READ, &hKey) != ERROR_SUCCESS)
+        return FALSE;
+
+    // query "extfile" key name
+    WCHAR szValue[64];
```
[the remark this answers is not in the document]
Comment (katahiromz) [reply dll/win32/shell32/dialogs/filetypes.cpp]: OK.

File: dll/win32/shell32/dialogs/filetypes.cpp [hunk 8]
```diff
@@ -0,0 +1,1780 @@
...
+        pEntry->hIconLarge = DoExtractIcon(pEntry, szPath, IconIndex, FALSE);
+        pEntry->hIconSmall = DoExtractIcon(pEntry, szPath, IconIndex, TRUE);
+
+        // update EDITTYPE_DIALOG
+        StringCbCopyW(pEditType->szIconPath, sizeof(pEditType->szIconPath), szPath);
+        pEditType->nIconIndex = IconIndex;
+
+        // set icon to dialog
+        HWND hwndIcon = GetDlgItem(hwndDlg, IDC_EDITTYPE_ICON);
+        SendMessageW(hwndIcon, STM_SETICON, (WPARAM)pEntry->hIconLarge, 0);
```
[the remark this answers is not in the document]
Comment (katahiromz) [reply dll/win32/shell32/dialogs/filetypes.cpp]: I turned to use SendDlgItemMessageW. OK.

File: dll/win32/shell32/dialogs/filetypes.cpp [hunk 9]
```diff
@@ -0,0 +1,1780 @@
...
+            // set default
+            StringCbCopyW(pEditType->szDefaultVerb, sizeof(pEditType->szDefaultVerb), action.szAction);
+            EnableWindow(GetDlgItem(hwndDlg, IDC_EDITTYPE_SET_DEFAULT), FALSE);
+            InvalidateRect(action.hwndLB, NULL, TRUE);
+            break;
+    }
+}
+
+static BOOL
+EditTypeDlg_ReadClass(HWND hwndDlg, PEDITTYPE_DIALOG pEditType, LPCWSTR ClassKey)
```
[the remark this answers is not in the document]
Comment (katahiromz) [reply dll/win32/shell32/dialogs/filetypes.cpp]: OK.

**Comment by katahiromz** [comment]
Done.
